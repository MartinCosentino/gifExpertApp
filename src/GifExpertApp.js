import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory'
import { GifGrid } from './components/GifGrid'



export const GifExpertApp = () => {


    const [category, setcategory] = useState(['WELCOME!'])

    // const handleAdd = () => {
    //     setcategory([...category, 'HunterXHunter']); //SPREAD para agregar elementos al arreglo usando setcategory para cambiar el state
    // }

    return (
        <>
            <h1 className="animate__animated animate__fadeIn" id="title" style={{ display: "flex", justifyContent: "center", fontSize: "10rem" }}>GifExpertApp</h1>
            <p className="animate__animated animate__backInDown" style={{ display: "flex", justifyContent: "center" }}>- Que Gif Estas Buscando? -</p> <br />

            <AddCategory setcategory={setcategory} />
            <hr />

            <ol>
                {
                    category.map(category => (
                        <GifGrid
                            key={category}
                            category={category}
                        />
                    ))
                    // <li key={category}>{category}</li> //key obligatorio y unico

                }
            </ol>

            <br />
            <div className="footerContenedor">
                <p className="footer"> Encontras lo que buscas? m√°s gifs en la pagina oficial de <a style={{ marginLeft: "4px" }} href="https://giphy.com/">  GIPHY </a>.</p>
            </div>

        </>
    )
}
