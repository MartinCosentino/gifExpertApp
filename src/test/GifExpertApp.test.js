import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import React from 'react';
import { GifExpertApp } from "../GifExpertApp";


describe('Preubas en el archivo principal GifExpertApp', () => {

    test('Debe de coincidir con el Snapshot', () => {

        const wrapper = shallow(<GifExpertApp />);

        expect(wrapper).toMatchSnapshot();



    });


});

