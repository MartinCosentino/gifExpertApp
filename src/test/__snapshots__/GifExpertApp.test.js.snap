// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preubas en el archivo principal GifExpertApp Debe de coincidir con el Snapshot 1`] = `
<Fragment>
  <h1
    id="title"
  >
    GifExpertApp
  </h1>
  <h4>
    Que Gif Estas Buscando?
  </h4>
  <AddCategory
    setcategory={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="WELCOME"
      key="WELCOME"
    />
  </ol>
</Fragment>
`;
